aws4.js

```diff
@@ -27,6 +27,11 @@ function RequestSigner(request, credentials) {
   this.request = request
   this.credentials = credentials || this.defaultCredentials()
 
+  // ES's hostParts are the other way round
+  // E.g. search-cluster-name-aaaa00aaaa0aaa0aaaaaaa0aaa.us-east-1.es.amazonaws.com
```

Comment:
I'm a little confused about this. There seem to be a couple of different hosts used by AWS ES:

http://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/what-is-amazon-elasticsearch-service.html#endpoints

That seems to suggest that the service name is actually `aes` when used with the `http://search-domainname-domainid.us-east-1.aes.amazonaws.com` type hostname, but `es` when used with a more standard `es.region.amazonaws.com` host name for configuration, etc.

Can you elaborate on this at all?